ch02/clients/ShipmentUpdateFunc.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Documents;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Spike.Inventory.Models;
+
+namespace clients
+{
+    public static class ShipmentUpdateFunc
+    {
+        [FunctionName("ShipmentUpdateFunc")]
+        public static async Task Run([CosmosDBTrigger(
+            databaseName: "inventory",
+            collectionName: "shipments",
+            ConnectionStringSetting = "CosmosDBConnection",
+            LeaseCollectionName = "leases")]IReadOnlyList<Document> input,
+            ILogger log, [DurableClient] IDurableOrchestrationClient starter)
+        {
+            
+            foreach (var item in input)
+            {
+                try
+                {
+                    //deserializes payload to object and "transform" it
+                    log.LogInformation(item.ToString());
+                    string stringifiedItem = item.ToString();
+                    InputShipmentEvent inputItem = new InputShipmentEvent();
+
+                    inputItem = JsonConvert.DeserializeObject<InputShipmentEvent>(stringifiedItem);
+                    var inventoryEvents = ResolveToInventoryEvent(inputItem);
+
+                    //re-serializes payload and sends it to orchestrator 
+                    foreach (var invEvent in inventoryEvents)
+                    {
+                        await starter.StartNewAsync("Orchestrator", null, invEvent);
+                    }
+                }
+                catch (Exception e)
+                {
+                    log.LogError(e.Message);
+                }
+            }
+        }
+        public static List<InventoryEvent> ResolveToInventoryEvent(InputShipmentEvent shipment)
+        {
+            List<InventoryEvent> eventList = shipment.items.Select(i => new InventoryEvent
+            {
+                Id = shipment.id,

```

Comment:
If you use `shipment.id` here each item will have the same id when passed to the orchestrator and the values will overwrite each other. They also won't have the correct id when written to mdc so they can't be queried

```suggestion
                Id = $"{shipment.divisionId}:{shipment.storeId}:{i.upc}",
```

Reply:
Excellent catch, @jcocchi. I didn't know that there was supposed to be any other value other than `shipment.id` here, so thanks!